framework/data.cpp

```diff
@@ -777,14 +777,17 @@ bool DataImpl::writeImage(UString systemPath, sp<Image> image, sp<Palette> palet
 {
 	auto outPath = fs::path(systemPath.str());
 	auto outDir = outPath.parent_path();
-	try
+	if (outDir != "" && outDir != ".")
```

Comment:
This would break for unexpected relative paths (e.g "./subfolder/..").

Perhaps using fs::exists(outDir) to avoid creating 'existing' directories (such as the "." in this example).

And maybe check if it exists, but fs::is_directory(outDir) is false, to give a better error if there happens to be a file of the same name we're trying to create a directory of?